node/webhooks.js

```diff
@@ -0,0 +1,125 @@
+                "currency_code": "MXN"
+            },
+            "payment_detail": {
+                "credentials": {
+                    "payment_method": {
+                        "method": "credit_or_debit_card",
+                        "display_name": null,
+                        "logo": null
+                    },
+                    "holacash_payment_token": null,
+                    "credit_or_debit_card": {
+                        "card_number": "424242XXXXXX4242",
+                        "expiration_month": "12",
+                        "expiration_year": "2024",
+                        "card_validation_code": "XXX"
+                    },
+                    "pay_by_store_type": null
+                },
+                "address": null,
+                "contact": null,
+                "name": null
+            },
+            "processing_instructions": {
+                "auto_capture": true
+            },
+            "purchase_details": {
+                "external_system_order_id": null,
+                "holacash_system_order_id": "096e06f5-ef85-4b54-b00a-e47eb0e94226",
+                "order_data": null
+            },
+            "consumer_details": {
+                "external_consumer_id": "9087321403udfs",
+                "address": null,
+                "contact": {
+                    "phone_1": null,
+                    "email": "test_user@gmail.com",
+                    "additional_contact_info": null
+                },
+                "name": {
+                    "first_name": "Test",
+                    "second_first_name": "Hola",
+                    "first_last_name": "Cash",
+                    "second_last_name": "User"
+                }
+            },
+            "shipping_details": null,
+            "additional_details": null
+        },
+        "related_transactions": null,
+        "additional_detail": null
+    },
+}
+
+// Split sign header into the timestamp and signature components
+const [timestamp, server_signature] = exampleSignature.split(",")
+
+// To generate the string to sign you have to concat the timestamp, a dot and the JSON.
+// The JSON should be a single line without spaces (The default behaviour of stringify function)
+const stringToSign = timestamp + "." + JSON.stringify(sampleJson)
+
+// The signature is done with HMAC_SHA256 algorithm and the key you can get from the portal (Exclusive for webhooks)
+// The digest is converted to a Hex string for debugging purposes. You can use the bytes digest for the compare.
+const client_signature = crypto.createHmac('sha256', exampleKey).update(stringToSign).digest("hex").toUpperCase()
+
+// Cryptographically compare the 2 signatures. In this case we use the timingSafeEqual function from the crypto lib in node
+// You can use any crypto comparison. Avoid at any costs comparing strings.
+const signs_are_equal = crypto.timingSafeEqual(Buffer.from(client_signature, "utf-8"), Buffer.from(server_signature, "utf-8"))
```

Comment:
@daniel-holacash can we make this into a function so that the client can copy/paste the function please? The function can return boolean (whether signature is valid or not)

Reply:
Done